utils/string.ts

```diff
@@ -4,16 +4,66 @@ export function kebabCase(name: string) {
 	return name.replace(KebabCaseRegex, (letter, pos) => (pos ? '-' : '') + letter.toLowerCase());
 }
 
-export function titleCase(str: string) {
-	// Hyphen and underscore.
-	str = str.replace(/(\-|_)/g, ' ');
+export type TitleCaseOptions = Partial<{
+	dropHyphens: boolean;
+	dropUnderscores: boolean;
+	expandCamelCase: boolean;
+	keepLcWords: boolean;
+}>;
 
-	// camelCase.
-	str = str.replace(/([a-z])([A-Z])/g, '$1 $2');
+// Common articles, coordinating conjunctions and prepositions that should normally be lowercased in titles.
+const lcWords = [
+	'A',
+	'An',
+	'The',
+	'And',
+	'But',
+	'For',
+	'Nor',
+	'Or',
+	'So',
+	'Yet',
+	'As',
+	'At',
+	'But',
+	'By',
+	'Etc',
+	'For',
+	'From',
+	'In',
+	'Into',
+	'Of',
+	'On',
+	'Onto',
+	'Than',
+	'To',
+	'Upon',
+	'Via',
+	'With',
+];
+
+export function titleCase(str: string, options?: TitleCaseOptions) {
```

Comment:
Should we get rid of multiple spaces in a row? `replace(/ +/, ' ')`

Reply:
I think so. This is meant to make a nice title, and multiple spaces aren't nice.
We shouldn't be running this function on user input like game titles anyways.

This comes in handy when we have intepolated strings in our translations that may be left empty, resulting in spaces on both sides for example.